### ISSUE TITLE
🐛 BUG: Regex in dynamic JSX expressions cause CompileError

### ISSUE THREAD

**Issue by mattjennings** [snowpackjs/astro#1059]
### What package manager are you using?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

When using a regular expression inside a dynamic JSX expression, such as `{content.replace(/"/g, '')}`, astro will fail with the following error:

```
[21:34:58] [snowpack] Build Result Error: There was a problem with a file build result.
[executing astro] CompileError [ParseError]: Expected }
    at error (/sandbox/node_modules/@astrojs/parser/src/utils/error.ts:40:15)
    at Parser.error (/sandbox/node_modules/@astrojs/parser/src/parse/index.ts:109:5)
    at Parser.eat (/sandbox/node_modules/@astrojs/parser/src/parse/index.ts:125:12)
    at mustache (/sandbox/node_modules/@astrojs/parser/src/parse/state/mustache.ts:403:12)
    at new Parser (/sandbox/node_modules/@astrojs/parser/src/parse/index.ts:55:15)
    at parse (/sandbox/node_modules/@astrojs/parser/src/parse/index.ts:227:18)
    at convertAstroToJsx (file:///sandbox/node_modules/astro/src/compiler/index.ts:31:15)
    at transformFromSource (file:///sandbox/node_modules/astro/src/compiler/index.ts:91:20)
    at compileComponent (file:///sandbox/node_modules/astro/src/compiler/index.ts:110:24)
    at Object.load (/sandbox/node_modules/astro/snowpack-plugin.cjs:68:28)
    at runPipelineLoadStep (/sandbox/node_modules/snowpack/lib/index.js:123895:32)
    at Object.buildFile (/sandbox/node_modules/snowpack/lib/index.js:124095:24)
    at /sandbox/node_modules/snowpack/lib/index.js:157676:37
    at FileBuilder.build (/sandbox/node_modules/snowpack/lib/index.js:157688:32)
    at Object.loadUrl (/sandbox/node_modules/snowpack/lib/index.js:177102:17)
    at load (/sandbox/node_modules/snowpack/lib/index.js:176699:28)
    at load (file:///sandbox/node_modules/astro/src/runtime.ts:106:17)
    at Server.<anonymous> (file:///sandbox/node_modules/astro/src/dev.ts:33:20) {
  start: { line: 11, column: 6, character: 147 },
  end: { line: 11, column: 6, character: 147 },
  filename: '/sandbox/src/pages/index.astro',
  frame: ` 9:     {content.replace(/"/g, '')}\n10:   </body>\n11: </html>\n          ^`,
  __snowpackBuildDetails: { name: 'snowpack-astro', step: 'load' }
}
```

### Steps to Reproduce

Create a .astro component with the following content:

```astro
---
import { Markdown } from 'astro/components';

const content = "\"Hello\""

---

<html>
  <body>
    {content.replace(/"/g, '')}
  </body>
</html>
```

It compiles fine if the .replace is moved up to the frontmatter

```astro
---
import { Markdown } from 'astro/components';

const content = "\"Hello\"".replace(/"/g, '')

---

<html>
  <body>
    {content}
  </body>
</html>
```

### Link to Minimal Reproducible Example (Optional)

https://codesandbox.io/s/staging-bash-il1d9?file=/src/pages/index.astro

**Comment by eyelidlessness** [comment]
I was tempted to laugh react the report but… I guess this is what you can expect with a whole new template language that has JS expressions mixed up with non-expressions.

**Comment by natemoo-re** [comment]
This is fixed in our new compiler, released under `astro@next`. See  https://stackblitz.com/edit/github-pbgyky?file=src%2Fpages%2Findex.astro

**Comment by natemoo-re** [comment]
For context, the team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
